DevTools/SourceMaps/explainer.md

```diff
@@ -0,0 +1,245 @@
+2. A value representing the absolute starting column number.
+3. A value representing the ending line, **relative to the starting line** value.
+4. A value representing the ending starting column number.
+5. An index into the `scopeNames` array.
+
+This enhancement chooses to use a combination of relative and absolute numbers because of the nature of the values in question. For source line offsets, using relative encoding makes sense, because the numbers are always expected to be increasing. For column numbers, however, it is highly likely that, because these values represent locations in the *source* file, offsets from starting- to ending-location would be negative, and further, that offsets from an outer-scoped function to an inner-scoped function would be negative. Given that one might expect most source lines to be less than 120 characters in length, there does not appear to be a major need to encode the column as relative.
+
+Given the above list, we should then encode the above values:
+
+```json
+{
+    "sources": ["example.js"],
+    "scopeNames": ["Global code", "foo"],
+    "scopes": [
+        "[0,0,0,33,0],[0,34,2,0,1],[2,1,1,6,0]"
+    ]
+}
+```
+
+Or, when encoded with VLQ, as:
+
+```json
+{
+    "sources": ["example.js"],
+    "scopeNames": ["Global code", "foo"],
+    "scopes": [
+        "AAAiCA,AkCEAC,ECCMA"
+    ]
+}
+```
+
+### Specific naming recommendations
+
+The following recommendations are **recommendations only** and not part of the "standard". They are, rather, a best attempt to identify how common examples can produce high-fidelity diagnostics experiences.
+
+* **Class names included**: Class member functions should indicate that they are members of classes. Although a common JavaScript convention has been to use `#` as a shorthand for `Class.prototype.member`, the inclusion of `#` as valid syntax for `private` field shorthand suggests that this should not be used. This spec recommends that the expansion `ClassName.functionName` should be used for instance functions, and `static ClassName.functionName` for static functions.
+* **Anonymous callback name inference**: Callback functions are often anonymous, but the name of the function being invoked is typically interesting. In the following example, an interesting function name might be something along the lines of `anonymous function passed to map`:
+
+```ts
+    return myArray.map((item, index) => {
+        /// ...
+    });
+```
+
+ * **Computed names expressions**: Computed names also present an interesting problem of data that are not available at build time. As best as possible, the original source text should be simply preserved:
+
+ ```ts
+class Example {
+    static [Symbol.species]() { return Object; }
+    ['to' + 'String']() {
+        return 'Hello, world.';
+    }
+}
+ ```
+
+* In the above example, we would want to see `static Example.[Symbol.species]` and `Example.['to' + 'String']` in the `scopeNames` field of the source map.

```

Comment:
Contrasting with pasta-sourcemaps (from [here](https://github.com/bloomberg/pasta-sourcemaps/blob/9d2022adbd87c76af1d10a352963ff1765c52def/src/parser.ts#L446-L456)):

Looks like it yields
- `<computed: identifier>`
- `<computed: stringLiteral>`
- `<computed: numericLiteral>`
- `<computed>` (else this)